src/main/java/org/isf/examination/rest/ExaminationController.java

```diff
@@ -203,65 +204,78 @@ public ResponseEntity<List<PatientExaminationDTO>> getByPatientId(@PathVariable
-            	throw new OHAPIException(new OHExceptionMessage("The minimum blood pressure must be lower than the maximum blood pressure"));
-            }
-        }
-        if (newPatientExamination.getPex_hr() != null) {
-            if (newPatientExamination.getPex_hr() < ExaminationParameters.HR_MIN || newPatientExamination.getPex_hr() > ExaminationParameters.HR_MAX) {
-            	throw new OHAPIException(new OHExceptionMessage("Heart rate should be between "+ExaminationParameters.HR_MIN +" and "+ExaminationParameters.HR_MAX));
-            }
-        }
-        if (newPatientExamination.getPex_temp() != null) {
-            if (newPatientExamination.getPex_temp() < ExaminationParameters.TEMP_MIN || newPatientExamination.getPex_temp() > ExaminationParameters.TEMP_MAX) {
-             	throw new OHAPIException(new OHExceptionMessage("The temperature should be between "+ExaminationParameters.TEMP_MIN +" and "+ExaminationParameters.TEMP_MAX));
-            }
-        }
-	if (newPatientExamination.getPex_sat() != null) {
-	    if (newPatientExamination.getPex_sat() < ExaminationParameters.SAT_MIN || newPatientExamination.getPex_sat() > ExaminationParameters.SAT_MAX) {
-	        throw new OHAPIException(new OHExceptionMessage("The saturation should be between "+ExaminationParameters.SAT_MIN+" and "+ExaminationParameters.SAT_MAX));
-	    } 
+
+	public void validateExamination(PatientExaminationDTO newPatientExamination) throws OHServiceException {
+		ExaminationParameters.initialize();
+
+		Integer pex_height = newPatientExamination.getPex_height();
+		Double pex_weight = newPatientExamination.getPex_weight();
+		if (pex_height == null || pex_weight == null) {
+			throw new OHAPIException(new OHExceptionMessage("The height and weight are compulsory"));
+		}
+		if (pex_height < ExaminationParameters.HEIGHT_MIN
+						|| pex_height > ExaminationParameters.HEIGHT_MAX) {
+			throw new OHAPIException(new OHExceptionMessage(
+							"The height should be between " + ExaminationParameters.HEIGHT_MIN + " and " + ExaminationParameters.HEIGHT_MAX));
+		}
+		if (pex_weight < ExaminationParameters.WEIGHT_MIN || pex_weight > ExaminationParameters.WEIGHT_MAX) {
+			throw new OHAPIException(new OHExceptionMessage(
+							"The weight should be between" + ExaminationParameters.WEIGHT_MIN + " and " + ExaminationParameters.WEIGHT_MAX));
+		}
+		Integer pex_ap_min = newPatientExamination.getPex_ap_min();
+		Integer pex_ap_max = newPatientExamination.getPex_ap_max();
+		if (pex_ap_min == null && pex_ap_max != null) {
+			throw new OHAPIException(new OHExceptionMessage("Malformed minimum/maximum blood pressure: minimum missing"));
+		}
+		if (pex_ap_min != null && pex_ap_max == null) {
+			throw new OHAPIException(new OHExceptionMessage("Malformed minimum/maximum blood pressure: maximum missing"));
+		}
+		if (pex_ap_min != null && pex_ap_max != null) {
+			if (pex_ap_min > pex_ap_max) {
```

Comment:
These two `if` statements could be joined.

Reply:
Done, thanks!